src/libstore/profiles.cc

```diff
@@ -150,27 +158,36 @@ void deleteGenerations(const Path & profile, const std::set<GenerationNumber> &
     }
 }
 
+/**
+ * Advanced the iterator until the given predicate `cond` returns `true`.
+ */
+static inline void iterDropUntil(Generations & gens, auto && i, auto && cond)
+{
+    for (; i != gens.rend() && !cond(*i); ++i);
+}
+
```

Comment:
I don't think this is worth factoring out. It's good to split those loops into functional units though.

Reply:
It is short, but I like how it makes it easy to use with `older` in the second function this is used.